torchdata/datapipes/map/util/unzipper.py

```diff
@@ -0,0 +1,73 @@
+# Copyright (c) Facebook, Inc. and its affiliates.
+from typing import Optional, Sequence, TypeVar
+
+from torch.utils.data.datapipes.datapipe import MapDataPipe
+
+
+T = TypeVar("T")
+
+
+# @functional_datapipe("unzip")  # This line must be kept for .pyi signature parser
```

Comment:
Why the `functional_datapipe` is commented out?

Reply:
`UnZipperMapDataPipe` actually doesn't implement the `MapDataPipe` protocol since it doesn't have a `__getitem__`. Its child `_UnZipperMapDataPipe` does. Therefore, it can't be registered directly.
The following line in the bottom must be used instead.
`MapDataPipe.register_datapipe_as_function("unzip", UnZipperMapDataPipe)`

Reply:
I think what `functional_datapipe` does is the same as `MapDataPipe.register_datapipe_as_function("unzip", UnZipperMapDataPipe)`. `functional_datapipe` is just a syntax sugar, right?
Like `UnZipperIterDataPipe`, we use the `functional_datapipe` as well even though it's just a DataPipe to return multiple `ChildDataPipe`. https://github.com/pytorch/data/blob/b6ade8f097bc9ac08460cd403034a35daff09cfa/torchdata/datapipes/iter/util/unzipper.py#L17-L18

Reply:
I double-checked and you are right. It works even when `UnZipper` doesn't have all the methods of `MapDataPipe`